packages/block-editor/src/components/iframe/index.js

```diff
@@ -303,47 +320,162 @@ function Iframe( {
 
 	useEffect( () => cleanup, [ cleanup ] );
 
-	const zoomOutAnimationClassnameRef = useRef( null );
-
-	// Toggle zoom out CSS Classes only when zoom out mode changes. We could add these into the useEffect
-	// that controls settings the CSS variables, but then we would need to do more work to ensure we're
-	// only toggling these when the zoom out mode changes, as that useEffect is also triggered by a large
-	// number of dependencies.
 	useEffect( () => {
-		if ( ! iframeDocument || ! isZoomedOut ) {
+		if (
+			! iframeDocument ||
+			// TODO: What should this condition be?
+			( scaleValue === 1 ) === ( prevScaleRef.current === 1 )
```

Comment:
The consumer is switching the `scale` prop between `1` and `'default'`, so maybe it should be tied to the `scale` prop instead of the value then? The current condition is effectively that, but probably less readable. I'm thinking instead it would probably be best to add just the `iframeDocument` and `scale` prop to the dependencies instead of everything would be sufficient.

Reply:
I think we keep it for now. It's simpler than refactoring the `useEffect`s.